conda_smithy/tests/test_feedstock_io.py

```diff
@@ -0,0 +1,710 @@
+
+
+class TestFeedstockIO_wo_Git(unittest.TestCase):
+    def setUp(self):
+        self.old_dir = os.getcwd()
+
+        self.tmp_dir = tempfile.mkdtemp()
+        os.chdir(self.tmp_dir)
+
+        with open(os.path.abspath(".keep"), "w") as fh:
+            fh.write("")
+
+
+    def test_repo_1(self):
+        self.assertTrue(fio.get_repo(self.tmp_dir) is None)
+
+
+    def test_repo_2(self):
+        self.assertTrue(fio.get_repo("") is None)
+
+
+    def test_get_mode_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        os.chmod(filename, set_mode)
+
+        file_mode = fio.get_mode_file(filename)
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_get_mode_file_2(self):
+        filename = "test.txt"
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        os.chmod(filename, set_mode)
+
+        file_mode = fio.get_mode_file(filename)
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_set_mode_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        fio.set_mode_file(filename, set_mode)
+
+        file_mode = os.stat(filename).st_mode
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_set_mode_file_2(self):
+        filename = "test.txt"
+
+        with open(filename, "w") as fh:
+            fh.write("")
+
+        perms = [
+            stat.S_IWUSR,
+            stat.S_IXUSR,
+            stat.S_IRUSR,
+            stat.S_IXGRP,
+            stat.S_IRGRP,
+            stat.S_IROTH
+        ]
+
+        set_mode = functools.reduce(op.or_, perms)
+
+        fio.set_mode_file(filename, set_mode)
+
+        file_mode = os.stat(filename).st_mode
+
+        self.assertEqual(file_mode & set_mode, set_mode)
+
+
+    def test_write_file_1(self):
+        filename = "test.txt"
+        filename = os.path.join(self.tmp_dir, filename)
+
+        write_text = "text"
+        with fio.write_file(filename) as fh:
+            fh.write(write_text)
+
+        read_text = ""
```

Comment:
True throughout. It's my C/C++ programmer side getting in the way. To use to thinking about scope in this way. Can drop them if you wish.